packages/launcher/src/renderer/launcher/contacts/InviteContactModal.js

```diff
@@ -341,50 +437,172 @@ class InviteContactModal extends Component<Props, State> {
+                w.balances.mft
+              }, ETH: ${w.balances.eth}`}</Text>
+            </AddContactDetailText>
+          </AddContactDetail>
+        ),
+      }
+    })
+
+    const titleSection = (
+      <Text
+        bold
+        variant="smallTitle"
+        color="#585858"
+        theme={{
+          display: 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+        }}>
+        {title}{' '}
+        {this.props.type === 'declineInvite' && (
           <Tooltip>
             <Text variant="tooltipTitle">
               Where did this address come from?
             </Text>
             <Text variant="tooltipText">
-              This is your default ETH address. You can change it in the Wallets
-              tab by selecting another address and designating it as default
-              instead.
+              This is/ was your default address at the time this contact sent
+              the blockchain invite.
             </Text>
           </Tooltip>
-        </Text>
-        <AddContactDetail border>
-          <Blocky>
-            <WalletIcon
-              address={this.props.user.profile.ethAddress || ''}
-              size="small"
-            />
-          </Blocky>
-          <AddContactDetailText>
-            <Text variant={['greyDark23', 'ellipsis', 'mono']} size={12}>
-              {this.props.user.profile.ethAddress}
+        )}
+        {this.props.type === 'retrieveStake' && (
+          <Tooltip>
+            <Text variant="tooltipTitle">
+              Where did this address come from?
             </Text>
-            {balanceLabel}
-          </AddContactDetailText>
-        </AddContactDetail>
-        {this.renderGasData()}
+            <Text variant="tooltipText">
+              This is the address you chose to stake MFT from at the time the
+              blockchain invite was sent.
+            </Text>
+          </Tooltip>
+        )}
+      </Text>
+    )
+
+    if (this.props.type === 'invite') {
+      return (
+        <Section>
+          {titleSection}
+          <DropDownContainer>
+            <DropDown
+              options={options}
+              valueKey="key"
+              displayKey="data"
+              name="walletDropdown"
+              defaultValue={this.props.selectedAddress}
+              onChange={addr => this.updateSelectedAddress(addr)}
+              variant={[this.state.dropdownError ? 'error' : '', 'maxWidth440']}
+              validation={feedback => this.validate(feedback)}
+              disabled={this.state.txProcessing}
+            />
+          </DropDownContainer>
+          {this.renderGasData()}
+        </Section>
+      )
+    } else {
+      return (
+        <Section>
+          {titleSection}
+          <AddContactDetail border>
+            {options[defaultIndex].data}
+          </AddContactDetail>
+          {this.renderGasData()}
+        </Section>
+      )
+    }
+  }
+
+  getButtonStatus = (firstAction, secondAction) => {
```

Comment:
This logic seems a bit strange, as it's only needed in one other method, and the second button seems only conditionally rendered. I think it might be easier to directly include this logic in the method using it rather than having this extra indirection.